src/components/common/Dialogs/EditDomainDialog/EditDomainDialog.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { Id } from '@colony/colony-js';
+import { string, object, number, boolean, InferType } from 'yup';
+
+import Dialog, { DialogProps, ActionDialogProps } from '~shared/Dialog';
+import { ActionHookForm as Form } from '~shared/Fields';
+
+import { ActionTypes } from '~redux/index';
+import { WizardDialogType } from '~hooks'; // useEnabledExtensions
+import { pipe, withMeta, mapPayload } from '~utils/actions';
+import { graphQlDomainColorMap } from '~types';
+import { DomainColor } from '~gql';
+import { notNull } from '~utils/arrays';
+
+import EditDomainDialogForm from './EditDomainDialogForm';
+import { getEditDomainDialogPayload } from './helpers';
+
+interface CustomWizardDialogProps extends ActionDialogProps {
+  filteredDomainId?: number;
+}
+
+type Props = DialogProps &
+  Partial<WizardDialogType<object>> &
+  CustomWizardDialogProps;
+
+const displayName = 'common.EditDomainDialog';
+
+const validationSchema = object()
+  .shape({
+    forceAction: boolean().defined(),
+    domainName: string().max(20).required(),
+    domainId: number().required(),
+    domainColor: number().defined(),
+    domainPurpose: string().max(90),
+    annotationMessage: string().max(4000),
+    motionDomainId: number(),
+  })
+  .defined();
+
+type FormValues = InferType<typeof validationSchema>;
+
+const EditDomainDialog = ({
+  callStep,
+  prevStep,
+  cancel,
+  close,
+  colony,
+  colony: { domains },
+  filteredDomainId,
+}: Props) => {
+  const colonyDomains = domains?.items.filter(notNull) || [];
+  const selectedDomain = colonyDomains.find((domain) =>
+    filteredDomainId === undefined || filteredDomainId === Id.RootDomain
+      ? domain.nativeId !== Id.RootDomain
+      : domain.nativeId === filteredDomainId,
+  );
+
+  const [isForce, setIsForce] = useState(false);
+  const navigate = useNavigate();
+
+  // const { isVotingExtensionEnabled } = useEnabledExtensions({
+  //   colonyAddress: colony.colonyAddress,
+  // });
+
+  const getFormAction = (actionType: 'SUBMIT' | 'ERROR' | 'SUCCESS') => {
+    const actionEnd = actionType === 'SUBMIT' ? '' : `_${actionType}`;
+
+    return !isForce // && isVotingExtensionEnabled
+      ? ActionTypes[`MOTION_DOMAIN_CREATE_EDIT${actionEnd}`]
+      : ActionTypes[`ACTION_DOMAIN_EDIT${actionEnd}`];
+  };
+
+  const transform = pipe(
+    mapPayload((payload) => getEditDomainDialogPayload(colony, payload)),
+    withMeta({ navigate }),
+  );
+
+  return (
+    <Form<FormValues>
+      defaultValues={{
+        forceAction: false,
```

Comment:
Does `forceAction` need to be included in the form's payload? It's only used to determine which saga to fire, I believe?

In which case, we could pass the `ForceToggle` component the setState function directly, meaning we wouldn't need to derive the `forced` state from the form state, which would mean we could remove lines 98-102

Reply:
It doesn't need to be included in the payload, but it's also not only used to determine which saga to fire as we have various conditionals inside the dialog forms that use this value, mainly revolving around permission stuff. Do we believe that in this case is it worth it to prop drill `setForceAction` and `forceAction`? 

FYI in case we prop drill those values instead of using the form context, currently, it would travel like this in almost every dialog that has an action that can be made into a motion:
```
Dialog > DialogForm > DialogHeading > ForceToggle
Dialog > DialogForm > DialogHeading > MotionDomainSelect (only forceAction)
Dialog > DialogForm > useDialogActionPermissions (only forceAction)
Dialog > DialogForm > DialogControls (only forceAction)
```

Reply:
Ok, let's leave as it is then. If it were just prop drilling to the ForceToggle component then I might've thought it was worth it but passing to multiple components probably isn't preferable to what we have currently. I was trying to think of a way to make the forced state accessible to the form's contents and the form itself. The only way I can think of would be to wrap the dialog in a context that stores the forced state, but that's just me thinking out loud (have not tested). Also that would be a separate issue if it did work. But mentioning just so you know what I was thinking.
![dialog](https://user-images.githubusercontent.com/64402732/219331575-b120a854-1918-48a4-b9ea-576d021189bf.png)

Reply:
Yeah, I was also thinking of ways of including this state in a more general component like the `ActionHookForm`, but I haven't encountered a good, easy way of doing it. Thank you for your thoughts! I will think about this some more